src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -25,9 +25,21 @@
 namespace Microsoft.AspNetCore.Authentication.OpenIdConnect
 {
     /// <summary>
-    /// A per-request authentication handler for the OpenIdConnectAuthenticationMiddleware.
+    /// A per-request authentication handler for OpenIdConnect.
     /// </summary>
-    public class OpenIdConnectHandler : RemoteAuthenticationHandler<OpenIdConnectOptions>
+    public class OpenIdConnectHandler : OpenIdConnectHandler<OpenIdConnectOptions>
+    {
+        public OpenIdConnectHandler(IOptionsSnapshot<OpenIdConnectOptions> options, ILoggerFactory logger, HtmlEncoder htmlEncoder, UrlEncoder encoder, ISystemClock clock)
+            : base(options, logger, htmlEncoder, encoder, clock)
+        { }
+    }
+
+    /// <summary>
+    /// A per-request authentication handler for OpenIdConnect.
+    /// </summary>
+    /// <typeparam name="TOptions">The options type used to configure the handler.</typeparam>
+    public class OpenIdConnectHandler<TOptions> : RemoteAuthenticationHandler<TOptions>
```

Comment:
If you decide to go with this approach, maybe it would be worth modifying all the authentication middleware to use the same pattern, for consistency?